### PR TITLE
framework/iotbus: fix iotbus coverity bugs.

### PR DESCRIPTION
**Pull request by an4967**
- fix layered architecture violation.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e584771bd65025869303c38a419412003ad9652e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e584771bd65025869303c38a419412003ad9652e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [89add7ca5312f70f3b71a0e526f8e57ecc681a3d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [89add7ca5312f70f3b71a0e526f8e57ecc681a3d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e796e6bb9f88b4651f4528e183e602a70a1f6d02] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e796e6bb9f88b4651f4528e183e602a70a1f6d02] - Code Rule Check OK.

File: os/drivers/iotdev/iotbus_sig.c [hunk 1]
```diff
@@ -110,14 +110,14 @@ static int iotbus_sig_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 
 		struct iotbus_int_info_s *data = (struct iotbus_int_info_s *)kmm_zalloc(sizeof(struct iotbus_int_info_s));
 		if (data == NULL) {
-			return IOTBUS_ERROR_OUT_OF_MEM;
+			return -EINVAL;
```
Comment (Taejun-Kwon) [inline os/drivers/iotdev/iotbus_sig.c]: ENOMEM

**Comment by seinfra** [comment]
Target : [22e66ebbcdeb565f71b8e8277d193d5a2fa098ee] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [22e66ebbcdeb565f71b8e8277d193d5a2fa098ee] - Code Rule Check OK.

File: os/drivers/iotdev/iotbus_sig.c (hunk 1 above)
Comment (an4967) [reply os/drivers/iotdev/iotbus_sig.c]: @Taejun-Kwon 
I fixed it. Thanks a lot.

**Pull request merged**